### PR TITLE
Introduce more consistent management of the local enablement cache

### PR DESCRIPTION
**Pull request by OddBloke**

### CODE REVIEW

**Comment by OddBloke** [comment]
Closed/reopened to trigger Travis. @blackboxsw, this is now ready for review.

File: uaclient/entitlements/tests/test_repo.py [hunk 1]
```diff
@@ -279,3 +295,33 @@ def test_enable_calls_adds_apt_repo_and_calls_apt_update(
             '/usr/share/keyrings/test.gpg')] == m_apt_add.call_args_list
         stdout, _ = capsys.readouterr()
         assert expected_output == stdout
+
+    @mock.patch.object(RepoTestEntitlement, 'setup_apt_config',
+                       return_value=True)
+    @mock.patch.object(RepoTestEntitlement, 'can_enable', return_value=True)
+    def test_enable_sets_public_local_enabled(
+            self, _m_can_enable, _m_setup_apt_config, entitlement):
+        # We patch the type of entitlement because packages is a property; we
```
Comment (blackboxsw) [inline uaclient/entitlements/tests/test_repo.py]: This is a very helpful comment. Thanks for that.

File: uaclient/tests/test_config.py [hunk 2]
```diff
@@ -266,3 +266,53 @@ def test_delete_cache_ignores_files_not_defined_in_data_paths(
...
+        with pytest.raises(Exception) as excinfo:
+            manager.set('test', non_bool_value)
+
+        expected_msg = 'LocalEnabledManager.set passed non-bool value'
+        assert expected_msg == str(excinfo.value)
+
+    def test_local_access_written_public(self, manager):
+        manager.set('test', True)
+
+        assert os.path.exists(
```
Comment (blackboxsw) [inline uaclient/tests/test_config.py]: Is it worth validation that not os.path.exists(manager._cfg.data_path('local-access', private=True)?

Comment (OddBloke) [reply uaclient/tests/test_config.py]: Yep, will add!

**Pull request merged**